wallet/src/de/schildbach/wallet/ui/SecurityViewModel.kt

```diff
@@ -18,54 +18,65 @@
 
 package de.schildbach.wallet.ui
 
-import androidx.lifecycle.LiveData
-import androidx.lifecycle.MutableLiveData
+import androidx.core.os.bundleOf
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
+import de.schildbach.wallet.Constants
+import de.schildbach.wallet.WalletApplication
 import kotlinx.coroutines.ExperimentalCoroutinesApi
-import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import org.bitcoinj.core.Coin
-import org.bitcoinj.utils.Fiat
+import org.bitcoinj.wallet.Wallet
 import org.dash.wallet.common.Configuration
+import org.dash.wallet.common.WalletDataProvider
 import org.dash.wallet.common.data.ExchangeRate
 import org.dash.wallet.common.services.ExchangeRatesProvider
+import org.dash.wallet.common.services.analytics.AnalyticsService
 import org.dash.wallet.common.util.GenericUtils
 import javax.inject.Inject
 
 @ExperimentalCoroutinesApi
 @HiltViewModel
-class SecurityViewModel @Inject constructor(var exchangeRates: ExchangeRatesProvider,
-                                            var configuration: Configuration)
-    : ViewModel() {
+class SecurityViewModel @Inject constructor(
+    private val exchangeRates: ExchangeRatesProvider,
+    private val configuration: Configuration,
+    private val walletData: WalletDataProvider,
+    private val analytics: AnalyticsService,
+    private val walletApplication: WalletApplication
+): ViewModel() {
+    private var selectedExchangeRate: ExchangeRate? = null
 
-    private val _selectedCurrencyCode = MutableStateFlow(configuration.exchangeCurrencyCode)
```

Comment:
Currency code isn't updated while we are on the `SecurityActivity`, so there is no need to observe it.